src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1571,7 +1571,8 @@ private static CSDiagnosticInfo CreateReflectionTypeNotFoundError(Type type)
         /// </summary>
         internal new NamedTypeSymbol? GetTypeByMetadataName(string fullyQualifiedMetadataName)
         {
-            return this.Assembly.GetTypeByMetadataName(fullyQualifiedMetadataName, includeReferences: true, isWellKnownType: false, conflicts: out var _);
+            return this.Assembly.GetTypeByMetadataName(fullyQualifiedMetadataName, includeReferences: true, isWellKnownType: false, conflicts: out var _,
+                ignoreCorLibraryDuplicatedTypes: this.Options.TopLevelBinderFlags.Includes(BinderFlags.IgnoreCorLibraryDuplicatedTypes));
```

Comment:
>ignoreCorLibraryDuplicatedTypes: this.Options.TopLevelBinderFlags.Includes(BinderFlags.IgnoreCorLibraryDuplicatedTypes [](start = 16, length = 118)

I am not sure what is the motivation for this change. It feels unexpected to me.